pkg/resources/model/reconciler.go

```diff
@@ -276,8 +276,9 @@ func NewValidationReconciler(
 			if req.IsEmptyPatch() {
 				continue
 			}
+			obj := req.Obj.DeepCopyObject()
```

Comment:
could address the reason for the copy with a comment
```suggestion
			obj := req.Obj.DeepCopyObject() // copy object so that the original is not changed by the call to Patch
```